### ISSUE TITLE
Best Practice for Grafana Updrade

### ISSUE THREAD

**Issue by shanky05** [grafana/grafana#3557]
Hello @torkelo 

Could you please suggest the best possible way to upgrade Grafana in Production Environment.

How do we ensure that grafana.db is not overwritten. It has both the changes , done by user and by default from product.

**Comment by torkelo** [comment]
make a backup of grafana.db is always good before upgrade. Then update grafana package and restart grafana-server. Grafana will automatically perform database schema upgrade.

**Comment by shanky05** [comment]
suppose there are 10 new dashboards added by the end user. Then from product side also  we have added some 5 dashboards. 

While upgrade a merge will be required right? what is the best way to do that?

**Comment by torkelo** [comment]
not sure what your talking about. merge? merge what? 

The dashboards you have created should still be there after the upgrade & restart of the server.

**Comment by shanky05** [comment]
hello Torkelo,

What i meant was suppose in 1st release which we did for Grafana inside our org  there were 10 dashboards. Now when end users used it they are added some more dashboards. Say 5 dashboards. Now in the next release, we also added some dashboard, say 3 dashboards.

Now the default dashboard count give by us ( product side  in our org) is 13  dashboards, now how do we upgrade making sure now grafana has all 13 ( default) + 5 ( end user) dashboard?
